nck/readers/twitter_reader.py

```diff
@@ -171,8 +179,8 @@ def __init__(
         # General inputs
         self.report_type = report_type
         self.entity = entity
-        self.start_date = start_date
-        self.end_date = end_date + timedelta(days=1)
+        self.start_date, self.end_date = build_date_range(start_date, end_date, date_range)
```

Comment:
Same for Twitter reader, we can probably remove the `self.validate_dates()` check below 😊

Reply:
Done 😄